dags/analyzer_helper/tests/integration/test_discord_transform_raw_data.py

```diff
@@ -0,0 +1,173 @@
+                "metadata": {
+                    "channel_id": "channel456",
+                    "thread_id": "thread123",
+                    "bot_activity": False,
+                },
+                "actions": [{"name": "message", "type": "emitter"}],
+                "interactions": [
+                    {
+                        "name": "reply",
+                        "users_engaged_id": ["user789"],
+                        "type": "emitter",
+                    }
+                ],
+            },
+            {
+                "author_id": "user789",
+                "date": self.period,
+                "source_id": "msg123",
+                "metadata": {
+                    "channel_id": "channel456",
+                    "thread_id": "thread123",
+                    "bot_activity": False,
+                },
+                "actions": [],
+                "interactions": [
+                    {
+                        "name": "reply",
+                        "users_engaged_id": ["user123"],
+                        "type": "receiver",
+                    }
+                ],
+            },
+        ]
+
+        result = self.transformer.transform(raw_data, self.platform_id, self.period)
+        self.assertEqual(result, expected_result)
+
+    def test_transform_data_with_user_mentions(self):
+        raw_data = [
+            {
+                "author": "user123",
+                "user_mentions": ["user456"],
+                "messageId": "msg123",
+                "channelId": "channel456",
+                "isGeneratedByWebhook": False,
+                "botActivity": False,
+                "threadId": "thread123",
+            }
+        ]
+
+        expected_result = [
+            {
+                "author_id": "user123",
+                "date": self.period,
+                "source_id": "msg123",
+                "metadata": {
+                    "channel_id": "channel456",
+                    "thread_id": "thread123",
+                    "bot_activity": False,
+                },
+                "actions": [{"name": "message", "type": "emitter"}],
+                "interactions": [
+                    {
+                        "name": "mention",
+                        "users_engaged_id": ["user456"],
+                        "type": "emitter",
+                    }
+                ],
+            }
+        ]
+
+        result = self.transformer.transform(raw_data, self.platform_id, self.period)
+        self.assertEqual(result, expected_result)
+
+    def test_transform_data_with_reactions(self):
+        raw_data = [
+            {
+                "author": "user123",
+                "reactions": ["user1, user2, :laugh:"],
+                "messageId": "msg123",
+                "channelId": "channel456",
+                "isGeneratedByWebhook": False,
+                "botActivity": False,
+                "threadId": "thread123",
+            }
+        ]
```

Comment:
The schema for rawinfos is in documentation 
https://rndadocs.notion.site/Analyzer-Discord-Pipeline-5009512d82a243deb35b3388e17b29c5#:~:text=An%20example%20for%20rawinfos%20document